fairscale/optim/oss.py

```diff
@@ -166,19 +167,22 @@ def step(self, closure: Optional[Callable[[], float]] = None, **kwargs: Any) ->
         OSS._sync_param_groups(self.param_groups, self.optim.param_groups)
 
         # Catch a possible change of devices in between OSS construction and step()
-        if self._default_device.type != self.param_groups[0]["params"][0].device.type:
-            logging.info("OSS detected that the parameter changed devices, re-allocating buffers")
-            self._clear_cache()
-            self.refresh_trainable()
+        with profiler.record_function("fairscale::oss::refresh_trainable"):
```

Comment:
@anj-s would that be ok ?

Reply:
I am going to modify offload to match this since it is definitely more succinct. Thanks!

Reply:
thanks to you @anj-s, it was a good call I think, some middle ground and following what others are doing, feels like a good outcome